### ISSUE TITLE
Date format is wrong

### ISSUE THREAD

**Issue by irevoire** [meilisearch/meilisearch#2172]
Our error message is wrong and we accept a format we shouldn’t accept.
The error message in question is:
```
expiresAt field value `xxxxxxxxx` is invalid. It should be in ISO-8601 format to represents a date or datetime in the future or specified as a null value. e.g. 'YYYY-MM-DD' or 'YYYY-MM-DDTHH:MM:SS'.
```

My problem is that...
```
YYYY-MM-DD
YYYY-MM-DDTHH:MM:SS
```

are not valid in ISO-8601. As you can see [here](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6) the RFC clearly state that a date time is composed of:
```
   date-time       = full-date "T" full-time
```
A `full-date` followed by a `T` and the followed by a `full-time`.
So obviously this format `YYYY-MM-DD` is wrong since there is no `T` at the end.

And now we’re going to look into the `full-time` thingy:
```
   full-time       = partial-time time-offset
   time-offset     = "Z" / time-numoffset
   time-numoffset  = ("+" / "-") time-hour ":" time-minute
```

Here as you can see a date either end by `Z` or by something complicated with a `+` or `-`.
Whatever, it never end by nothing like that `YYYY-MM-DDTHH:MM:SS`.

**Comment by irevoire** [comment]
So, it look like I was a bit confused between ISO 8601 and the RFC 3339 which are not the same: https://ijmacd.github.io/rfc3339-iso8601/
Currently we’re only handling the rfc3339.
Also `YYYY-MM-DD` is a valid date (but not a datetime).
And `YYYY-MM-DDTHH:MM:SS` is invalid but we can handle it to help people.

Since the rfc3339 also handle `YYYY-MM-DD HH:MM:SSZ` I would like to handle `YYYY-MM-DD HH:MM:SS` which is easier to write and read.

I think we just need to update the error message to clearly state what we handle (the RFC3339) and show the helper with a space.
I’ll do it in #2173

**Comment by curquiza** [comment]
@meilisearch/docs-team you are concerned by this issue
TLDR; we are not using ISO-8601 for our date format but RFC 3339 :)
